microservices/download/utils.js

```diff
@@ -28,12 +58,16 @@ function query_type(req) {
 
 async function displayError(req, res, errMsg) {
   if (errMsg.code && errMsg.msg) {
-    res.status(errMsg.code).json(errMsg.msg);
-    return;
+
+    res.writeHead(errMsg.code, { "Content-Type": "application/json" });
+    res.end(JSON.stringify({ message: errMsg.msg }));
+    res.end();
```

Comment:
Am I mistaken that calling `res.end()` multiple times is redundant?

I wouldn't be surprised if there is a reason for this... This is JavaScript, after all.

(Same thing for line 69 + line 70).

Reply:
[COMMENTARY, no need to read or address this, optional to view]: If I had to guess, you initially wrote it as `res.write()` but refactored the post content into `res.end()`? I know I looked into that myself for pr #49. I wasn't able to get it to work, for some reason.